packages/compiler/src/experimental/typekit/define-kit.ts

```diff
@@ -0,0 +1,81 @@
+import { type Program } from "../../core/program.js";
+
+let currentProgram: Program | undefined;
+
+/** @experimental */
+export function setCurrentProgram(program: Program): void {
```

Comment:
hhm, is that not potentially going to cause issue with concurrent compilation?

Reply:
It shouldn't cause problems between emitters as they'd try to set the same value. We haven't seen issues in tests so far either. I'll be happy to try out a repro if you have something in mind